app/assets/javascripts/controllers/network_router/network_router_form_controller.js

```diff
@@ -1,35 +1,78 @@
-ManageIQ.angular.app.controller('networkRouterFormController', ['$http', '$scope', 'networkRouterFormId', 'miqService', function($http, $scope, networkRouterFormId, miqService) {
+ManageIQ.angular.app.controller('networkRouterFormController', ['networkRouterFormId', 'miqService', 'API', function(networkRouterFormId, miqService, API) {
   var vm = this;
 
-  vm.networkRouterModel = {
-    name: '',
-    cloud_subnet_id: '',
+  var init = function() {
+    vm.afterGet = false;
+
+    vm.networkRouterModel = {
+      name: '',
+      extra_attributes: { external_gateway_info: { enable_snat: true } },
+      cloud_subnet_id: null,
+      external_gateway: false,
+    };
+
+    vm.formId = networkRouterFormId;
+    vm.model = "networkRouterModel";
+    vm.newRecord = networkRouterFormId === "new";
+    vm.saveable = miqService.saveable;
+
+    if (vm.newRecord) {
+      vm.afterGet = true;
+      vm.modelCopy = angular.copy(vm.networkRouterModel);
+    } else {
+      miqService.sparkleOn();
+      API.get("/api/network_routers/" + networkRouterFormId + "?attributes=name,ems_id,admin_state_up,cloud_network_id,extra_attributes,cloud_tenant,ext_management_system,cloud_subnets").then(function(data) {
+        vm.networkRouterModel = _.omit(data, [ 'cloud_tenant', 'ext_management_system' ]);
+        vm.networkRouterModel.ext_management_system = {
+          id: data.ext_management_system.id,
+          name: data.ext_management_system.name,
+        };
+        vm.networkRouterModel.cloud_tenant = { id: data.cloud_tenant.id,  name: data.cloud_tenant.name, };
+
+        if (data.extra_attributes.external_gateway_info && ! _.isEmpty(data.extra_attributes.external_gateway_info)) {
+          vm.networkRouterModel.external_gateway = true;
+          return getCloudSubnetsByRef(vm.networkRouterModel.extra_attributes.external_gateway_info.external_fixed_ips[0].subnet_id);
+        } else {
```

Comment:
Code Climate's warning `Unnecessary 'else' after 'return'`. It means that anything in `else` should be instead out of that condition. Something like this:

```
if (...) {
   return ...;
}
vm.networkRouterModel.external_gateway = false;
```

Reply:
Done